src/execution/operator/projection/physical_tableinout_function.cpp

```diff
@@ -68,7 +70,14 @@ OperatorResultType PhysicalTableInOutFunction::Execute(ExecutionContext &context
 	TableFunctionInput data(bind_data.get(), state.local_state.get(), gstate.global_state.get());
 	if (projected_input.empty()) {
 		// straightforward case - no need to project input
-		return function.in_out_function(context, data, input, chunk);
+		auto result = function.in_out_function(context, data, input, chunk);
+		if (function.ordinality_data.ordinality_request == Ordinality_request_t::REQUESTED) {
+			const idx_t ordinality = chunk.size();
+			function.ordinality_data.SetOrdinality(chunk, gstate.current_ordinality_idx, ordinality);
```

Comment:
We are reading `gstate.current_ordinality_idx` without a lock, whereas we are modifying it below with a lock - one of these is wrong. I suspect we need to move this to the thread-local state anyway however, in which case we don't need a lock at all.